design/physical-network-api-prototype.md

```diff
@@ -0,0 +1,184 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# prototype-a-physical-network-api
+
+## Status
+
+implementable
+
+## Table of Contents
+
+<!--ts-->
+   * [Title](#title)
+      * [Status](#status)
+      * [Table of Contents](#table-of-contents)
+      * [Summary](#summary)
+      * [Motivation](#motivation)
+         * [Goals](#goals)
+         * [Non-Goals](#non-goals)
+      * [Proposal](#proposal)
+         * [User Stories [optional]](#user-stories-optional)
+            * [Story 1](#story-1)
+            * [Story 2](#story-2)
+         * [Implementation Details/Notes/Constraints [optional]](#implementation-detailsnotesconstraints-optional)
+         * [Risks and Mitigations](#risks-and-mitigations)
+      * [Design Details](#design-details)
+         * [Work Items](#work-items)
+         * [Dependencies](#dependencies)
+         * [Test Plan](#test-plan)
+         * [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+         * [Version Skew Strategy](#version-skew-strategy)
+      * [Drawbacks [optional]](#drawbacks-optional)
+      * [Alternatives [optional]](#alternatives-optional)
+      * [References](#references)
+
+<!-- Added by: stack, at: 2019-02-15T11:41-05:00 -->
+
+<!--te-->
+
+## Summary
+
+The Metal³ project is currently centered around an API for managing physical
+hosts.  This proposal is to explore expanding the scope to also include an API
+for managing physical network devices.  This exploration would be done by
+starting with a prototype that can configure some aspects of a top-of-rack
+(ToR) switch.
+
+## Motivation
+
+Metal³ follows the paradigm of Kubernetes Native Infrastructure (KNI), which is
+an approach to use Kubernetes to manage underlying infrastructure.  Managing
+the configuration of some physical network devices is closely related to
+managing physical hosts.
+
+As bare metal hosts are provisioned or later repurposed, there may be
+corresponding physical network changes that must be made, such as reconfiguring
+a ToR switch port.  If the provisioning of the physical host is managed through
+a Kubernetes API, it would be convenient to be able to reconfigure related
+network devices using a similar API.
+
+### Goals
+
+Produce a `physical-network-api` prototype, which includes the following:
+
+* A `Switch` CRD with one or more fields that corresponding to switch
```

Comment:
I think for this prototype approaching it as a Switch is fine. I think for future work it would be good to consider whether we can abstract that further to provide an API that focuses on connectivity, and not implementation details of switches.